app/ui/browser/views/navbar/location.jsx

```diff
@@ -140,6 +132,19 @@ export class Location extends Component {
     if (this.state.showURLBar && document.activeElement !== this.refs.input) {
       this.refs.input.focus();
     }
+
+    // this happens when userTypedLocation changed outside of a 'normal' input
+    // change i.e. 'paste' context menu.  Need to make sure the input is actually set.
+    // No tests for this right now, since shallow rendering doesn't support 'refs'.
+    const nextLocation = this.getVisibleLocation();
+    if (this.refs.input && nextLocation && this.refs.input.value !== nextLocation) {
+      this.setInputValue(nextLocation);
+    }
+  }
+
+  getVisibleLocation() {
```

Comment:
both `visible` and `location` are vague in this context -- maybe something like `getRenderURL`, or atleast a comment